src/core/analytics.js

```diff
@@ -789,19 +806,22 @@ class Analytics {
         });
       }
 
-      // convert integrations object to server identified names, kind of hack now!
-      transformToServerNames(rudderElement.message.integrations);
-      rudderElement.message.integrations = getMergedClientSuppliedIntegrations(
-        this.integrationsData,
-        clientSuppliedIntegrations,
-      );
+      // Holding the cloud mode events based on flag and integrations load check
+      if (!this.options?.bufferDataPlaneEventsUntilReady || this.clientIntegrationObjects) {
```

Comment:
Sanitizing this option is required. Refer to Ln 1092-1094.

Reply:
updated